src/containers/AdminCampaignEdit.jsx

```diff
@@ -202,17 +206,37 @@ export class AdminCampaignEdit extends React.Component {
     // 3. Refetch/poll updates data in loadData component wrapper
     //    and triggers *this* method => this.props.campaignData => this.state.campaignFormValues
     // So campaignFormValues should always be the diffs between server and client form data
-    let { expandedSection, isPolling } = this.state;
+    let { expandedSection } = this.state;
     let expandedKeys = [];
     if (expandedSection !== null) {
       expandedSection = this.sections()[expandedSection];
       expandedKeys = expandedSection.keys;
     }
 
+    const keysInUnsavedSections = [];
+    this.sections().forEach(section => {
+      if (!this.checkSectionSaved(section)) {
+        keysInUnsavedSections.push(...section.keys);
+      }
+    });
+
+    const newCampaignData = newProps.campaignData.campaign;
+    const changedKeysInUnsavedSections = [];
```

Comment:
are these *changed keys* or just keys that we don't want to overwrite due to unsaved info?  If the latter, then let's change the variable name.

Reply:
Added some comments and renamed the variable. Hopefully this makes my intention clear(er).

Reply:
what are the keys here?  Aren't a bunch of them objects and not values (e.g. Interaction Steps array, which is one of the frequent issues)